src/components/post/ViewPost.jsx

```diff
@@ -29,6 +29,28 @@ export const ViewPost = () => {
     setLoading(false);
   };
 
+  const query = async () => { 
```

Comment:
You can't call a function outside of a useEffect, you have to define everything inside (so the way it was before).
useEffect has a dependency array where you put all the state and function you use. If you put the query function in the dependency array, you'll end up with a loop. You call the query function, then useEffect see something change about this function so launch the useEffect again and again and again.

Not adding your dependency can lead to big state problem.

There is one way to do it, add a 'useCallback' so that the result from the query is cached, but this is outside our need for this.
Check this https://devtrium.com/posts/async-functions-useeffect